### PR TITLE
Change Action to include the context's exec env

### PR DESCRIPTION
**Pull request by rgrinberg**

### CODE REVIEW

File: src/action.ml [hunk 1]
```diff
@@ -735,7 +735,16 @@ let rec exec t ~ectx ~dir ~env ~stdout_to ~stderr_to =
   | Run (Error e, _) ->
     Prog.Not_found.raise e
   | Run (Ok prog, args) ->
-    exec_run ~ectx ~dir ~env ~stdout_to ~stderr_to prog args
+    exec_run ~ectx ~dir ~env:(
+      match ectx.context with
```
Comment (diml) [inline src/action.ml]: We shouldn't need this, `even` is initialized with `context.env` already

Comment (rgrinberg) [reply src/action.ml]: Hmm yeah, I suppose we should also allow the user to override the ENV set here.

**Comment by diml** [comment]
I'm just not entirely sure what we should do for x-compilation, thought to be consistent with the way we interpret `(run ...)`, we might want to set at least `PATH` to the host context one

**Comment by rgrinberg** [comment]
> I'm just not entirely sure what we should do for x-compilation, thought to be consistent with the way we interpret (run ...), we might want to set at least PATH to the host context one

Just for `run`? What about `system` and `bash`?

**Pull request merged**